cpp/velox/shuffle/VeloxSortShuffleWriter.cc

```diff
@@ -278,7 +278,7 @@ uint32_t VeloxSortShuffleWriter::maxRowsToInsert(uint32_t offset, uint32_t rows)
 }
 
 void VeloxSortShuffleWriter::acquireNewBuffer(int64_t memLimit, uint64_t minSizeRequired) {
-  auto size = std::max(std::min((uint64_t)memLimit >> 2, 64UL * 1024 * 1024), minSizeRequired);
+  auto size = std::max(std::min<uint64_t>(memLimit >> 2, 64UL * 1024 * 1024), minSizeRequired);
```

Comment:
we still need to cast memLimit to uint64_t from int64_t for >>.

Can you check if we can change the memLimit type to uint64_t?

Reply:
IIUC, memLimit should always be a positive number.

Reply:
can you change the data type to uint64_t in the call chains.

Reply:
`auto size = std::max(std::min<uint64_t>((uint64_t)memLimit >> 2, 64UL * 1024 * 1024), minSizeRequired);`
  Is this so?

Reply:
or 
`auto size = std::max(std::min((uint64_t)memLimit >> 2, (uint64_t)64UL * 1024 * 1024), minSizeRequired);`

Reply:
No, I mean VeloxSortShuffleWriter::acquireNewBuffer(**uint64_t** memLimit, uint64_t minSizeRequired)

Also change the data type where acquireNewBuffer is called.